### PR TITLE
update DMTCP checkpoint demo code

### PR DESCRIPTION
**Pull request by JasonYangShadow**
## Description of the Pull Request (PR):

update DMTCP code to patch the demo. 

## This fixes or addresses the following GitHub issues:

- Fixes https://github.com/apptainer/apptainer/issues/2325

### CODE REVIEW

**Comment by JasonYangShadow** [comment]
https://github.com/CrossNox/m2r2/issues/68
it looks like building docs fails due to the issue listed above.

File: checkpoint.rst [hunk 1]
```diff
@@ -168,21 +159,19 @@ the state of a variable on the server.
...
 
-    $ curl localhost:8888; echo
-    0
+    $ curl --http0.9 localhost:8888
+    request:1
 
-We can see that it is set to ``0`` by default when this application is started
-normally. We can now update the state of the server from ``0`` to ``1`` with
-the following ``POST`` request:
+We can see that request count value is ``1`` when this application is started
```
Comment (DrDaveD) [inline checkpoint.rst:165]: Add `the` before `request`

File: checkpoint.rst [hunk 2]
```diff
@@ -168,21 +159,19 @@ the state of a variable on the server.
...
 
-    $ curl -X POST localhost:8888 -d '1'; echo
-    1
-    $ curl localhost:8888; echo
-    1
+    $ curl --http0.9 localhost:8888
+    request:2
 
-Now that variable on our server is in a new state, ``1``, we can use the
+Now that request count variable on our server is in a new state, ``2``, we can use the
```
Comment (DrDaveD) [inline checkpoint.rst:174]: Again, `the` before `request

File: checkpoint.rst [hunk 3]
```diff
@@ -210,14 +199,28 @@ our application's state:
     INFO:    instance started successfully
 
 
-And now we can verify the variable on the server has been properly restored to
-a value of ``1``, instead of the default of ``0``:
+And now when we get access to the application again, the request count value is ``3`` as expected,
+meaning that previous request count value is ``2``.
```
Comment (DrDaveD) [inline checkpoint.rst:203]: Change `is` to `was` and either add `the` before `previous` or change `previous` to `previously the`.

File: checkpoint.rst [hunk 4]
```diff
@@ -210,14 +199,28 @@ our application's state:
...
 
 .. code::
 
-    $ curl localhost:8888; echo
-    1
+    $ curl --http0.9 localhost:8888
+    $ request:3
+
+We can repeat previous two steps, i.e. stop server instance and restart it via dmtcp to verify the restoration
+of the value of request count.
```
Comment (DrDaveD) [inline checkpoint.rst:211]: Add `the` before `previous`,`server`, and `request`

File: checkpoint.rst [hunk 5]
```diff
@@ -210,14 +199,28 @@ our application's state:
...
+
+.. code::
+
+    $ {command} instance stop server
+    $ {command} instance start --dmtcp-restart example-checkpoint server.sif restarted-server 8888
+
+Then access the application and see that request count value is restored as expected.
```
Comment (DrDaveD) [inline checkpoint.rst:218]: Add `the` before `request`

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Comment by DrDaveD** [comment]
> [CrossNox/m2r2#68](https://github.com/CrossNox/m2r2/issues/68) it looks like building docs fails due to the issue listed above.

Can you fix it by installing a docutils version <0.21 in the github action?  It looks like it must currently be installing docutils as an implied dependency because it isn't explicitly listed.

File: .github/workflows/pull_request.yml [hunk 6]
```diff
@@ -34,7 +34,7 @@ jobs:
 
       - name: Install Sphinx
         run: |
-          pip install --user --upgrade --upgrade-strategy eager setuptools sphinx sphinx-rtd-theme restructuredtext_lint rstcheck pygments m2r2
+          pip install --user --upgrade --upgrade-strategy eager docutils==0.16 setuptools sphinx sphinx-rtd-theme restructuredtext_lint rstcheck pygments m2r2
```
Comment (DrDaveD) [inline .github/workflows/pull_request.yml:37]: Please add a comment here explaining why docutils version is selected and pointing to the m2r2 issue.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Review by DrDaveD — APPROVED** [review]

**Pull request merged**